src/adapter/breakpoints.ts

```diff
@@ -449,8 +464,10 @@ export class BreakpointManager {
       await Promise.all(previous.map(b => b.remove()));
     }
     this._totalBreakpointsCount += breakpoints.length;
-    if (this._thread) breakpoints.forEach(b => b.set(this._thread!));
-    this._updateSourceMapHandler();
+    if (this._thread) {
+      breakpoints.forEach(b => b.set(this._thread!));
+      this._updateSourceMapHandler(this._thread);
```

Comment:
It seems like you don't need to call this method here anymore, since it does not depend on `this._totalBreakpointsCount`.

Note that it was added to avoid slowdown for the browser scenario where predictor is not always available, and user launches without any breakpoints often enough so we can skip extra pauses. I think this might still be a valid concern.

Reply:
Ah, yea, I see. I didn't mentally parse this code before. It seems to work fine without that line. It doesn't look like we ever _don't_ try to install the source map handler, but it's probably safe to do so when we know there aren't any breakpoints set. I'll look at that in a followup PR :)